### PR TITLE
Individual MO residential address displayed to the user

### PR DESCRIPTION
**Pull request by bmcgreevyCH**
JIRA link
https://companieshouse.atlassian.net/browse/UAR-619

Change description
Individual MO residential address displayed to the user

### CODE REVIEW

File: src/services/overseas-entities/types.ts [hunk 1]
```diff
@@ -313,6 +313,58 @@ export interface ManagingOfficerCorporateResource {
     resigned_on?: string
 }
 
+export interface ManagingOfficerDataResource {
+    managingOfficerAppointmentId: string;
```
Comment (AndreKramer7654) [inline src/services/overseas-entities/types.ts:317]: need to add hashed id here

File: src/services/overseas-entities/types.ts [hunk 2]
```diff
@@ -313,6 +313,58 @@ export interface ManagingOfficerCorporateResource {
...
+    residentialAddress: privateADDRESS;
+    principalAddress: privateADDRESS;
+    dateOfBirth: string;
+    contactNameFull: string;
+    contactEmailAddress: string;
+    hashedId: string;
+}
+
+export interface privateADDRESS {
```
Comment (AndreKramer7654) [inline src/services/overseas-entities/types.ts:356]: PrivateAddress

File: src/services/overseas-entities/types.ts [hunk 3]
```diff
@@ -313,6 +313,58 @@ export interface ManagingOfficerCorporateResource {
...
+export interface ManagingOfficersPrivateDataResource {
+    mo_private_data: ManagingOfficerDataResource[];
+}
+
+export interface ManagingOfficersPrivateData {
+    moPrivateData: ManagingOfficerData[];
+}
+
+export interface ManagingOfficerData {
+    managingOfficerAppointmentId: string;
```
Comment (AndreKramer7654) [inline src/services/overseas-entities/types.ts:347]: need hashed id too.

File: src/services/overseas-entities/types.ts [hunk 4]
```diff
@@ -313,6 +313,58 @@ export interface ManagingOfficerCorporateResource {
...
+    residentialAddress: privateADDRESS;
+    principalAddress: privateADDRESS;
+    dateOfBirth: string;
+    contactNameFull: string;
+    contactEmailAddress: string;
+    hashedId: string;
```
Comment (AndreKramer7654) [inline src/services/overseas-entities/types.ts:353]: minor - I missed the hashedId as added at end. May be better named hashManagingOfficerAppointmentId and moved up?

File: src/services/overseas-entities/types.ts (hunk 1 above)
Comment (AndreKramer7654) [inline src/services/overseas-entities/types.ts:317]: the managingOfficerAppointmentId should be optional (using managingOfficerAppointmentId?: string) or removed. 
The other PR has everything optional.

File: test/services/overseas-entities/overseas.entities.mock.ts [hunk 5]
```diff
@@ -47,6 +47,28 @@ export const ADDRESS: Address = {
     postcode: "BY 2"
 };
 
+export const ADDRESS_API = {
```
Comment (n-r-b) [inline test/services/overseas-entities/overseas.entities.mock.ts:50]:
```suggestion
export const PRIVATE_ADDRESS_MOCK = {
```

File: test/services/overseas-entities/overseas.entities.mock.ts [hunk 6]
```diff
@@ -47,6 +47,28 @@ export const ADDRESS: Address = {
...
+    poBox: "PO Box information",
+    postalCode: "postal code information",
+    premises: "premises information",
+    region: "region information"
+};
+
+export const MANAGING_OFFICERS_DATA_MOCK = {
+    managingOfficerAppointmentId: "123456789",
+    residentialAddress: ADDRESS_API,
+    principalAddress: ADDRESS_API,
```
Comment (n-r-b) [inline test/services/overseas-entities/overseas.entities.mock.ts:65]: Assuming these need to be updated too?
```suggestion
    residentialAddress: PRIVATE_ADDRESS_MOCK,
    principalAddress: PRIVATE_ADDRESS_MOCK,
```

**Review by n-r-b — APPROVED** [review]

**Pull request merged**